packages/react-instantsearch-hooks-dom/src/widgets/RefinementList.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { useRefinementList } from 'react-instantsearch-hooks';
+
+import { cx } from '../ui/lib/cx';
+import { RefinementList as RefinementListUiComponent } from '../ui/RefinementList';
+import { SearchBox as SearchBoxUiComponent } from '../ui/SearchBox';
+import { ShowMoreButton } from '../ui/ShowMoreButton';
+
+import type { RefinementListProps as RefinementListUiComponentProps } from '../ui/RefinementList';
+import type { RefinementListItem } from 'instantsearch.js/es/connectors/refinement-list/connectRefinementList';
+import type { RefinementListWidgetParams } from 'instantsearch.js/es/widgets/refinement-list/refinement-list';
+import type { UseRefinementListProps } from 'react-instantsearch-hooks';
+
+export type RefinementListProps = Omit<
+  RefinementListUiComponentProps,
+  'items' | 'onRefine' | 'searchBox' | 'noResults' | 'showMoreButton'
+> &
+  UseRefinementListProps &
+  Pick<RefinementListWidgetParams, 'searchable' | 'searchablePlaceholder'>;
+
+export function RefinementList({
+  searchable,
+  searchablePlaceholder,
+  attribute,
+  operator,
+  limit,
+  showMore,
+  showMoreLimit,
+  sortBy,
+  escapeFacetValues,
+  transformItems,
+  ...props
+}: RefinementListProps) {
+  const {
+    canToggleShowMore,
+    isFromSearch,
+    isShowingMore,
+    items,
+    refine,
+    searchForItems,
+    toggleShowMore,
+  } = useRefinementList(
+    {
+      attribute,
+      operator,
+      limit,
+      showMore,
+      showMoreLimit,
+      sortBy,
+      escapeFacetValues,
+      transformItems,
+    },
```

Comment:
we can change this in a different PR, but I wonder if we should pass all props to useRefinementList too. We know they won't overlap, as it's very unlikely for dom props, but it would mean a new parameter could be passed immediately, without code change

Reply:
We're inlining the DOM props onto the UI component, so we need to separate DOM props from widget props anyway.